### PR TITLE
update readme and templates to reflect changes in datapipes

### PR DESCRIPTION
**Pull request by zakwatts**
PR mainly of edits to the documentation and template configs.

Commit history is very large and was causing issues when downloading, even with a GitHub buffer size increased to 1Gb I was running into download problems. Hence it has suggested that a clone depth can be used with cloning the repo.

Instructions on how to set up and get started with ocf_datapipes in order to create batches have been added alongside a more step by step guide through where to specify config files and google cloud set up instructions (with gcsfc being adding the requirements.txt)

Recent changes in ocf_datapipes also meant that the example configs were out of date, with multiple NWPs now being used the way the nwps in the config were set up have changed slightly (by an indentation, I believe to reflect it now being a dictionary of NWPs but I have not looked into this too much!)

### CODE REVIEW

**Review by peterdudfield — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -69,116 +83,104 @@ https://huggingface.co/datasets/openclimatefix/dwd-icon-eu which includes the UK
...
+
+To satellite data hosted by Google:
+
+```
+satellite:
+    satellite_zarr_paths:
+      - "gs://public-datasets-eumetsat-solar-forecasting/satellite/EUMETSAT/SEVIRI_RSS/v4/2020_nonhrv.zarr"
+      - "gs://public-datasets-eumetsat-solar-forecasting/satellite/EUMETSAT/SEVIRI_RSS/v4/2021_nonhrv.zarr"
+```
+Datapipes is currently set up to use 12 channels from the satellite data which is the YEAR_nonhrv.zarr rather than YEAR_hrv.zarr.
```
Comment (jacobbieker) [inline README.md:181]: 11 channels, the 12th is HRV, which is not included in these ones.

File: README.md [hunk 2]
```diff
@@ -69,116 +83,104 @@ https://huggingface.co/datasets/openclimatefix/dwd-icon-eu which includes the UK
...
+
+```
+python scripts/save_batches.py datamodule=streamed_batches +batch_output_dir="./output" +num_train_batches=10 +num_val_batches=5
+```
+
+In this function the datamodule argument looks for a config under `PVNet/configs/datamodule`. The examples here are either to use "premade_batches" or "streamed_batches".
+
+Its important that the dates set for the training, validation and testing in the datamodule (`streamed_batches.yaml`) config are within the ranges of the dates set for the input features in the configuration (`gcp_configuration.yaml`).
+
+If downloading data from gcp bucket or satellite data make sure to authenticate gcloud:
```
Comment (jacobbieker) [inline README.md:159]:
```suggestion
If downloading private data from gcp bucket make sure to authenticate gcloud (the public satellite data does not need authentication):
```

File: configs.example/datamodule/configuration/template_configuration.yaml [hunk 3]
```diff
@@ -15,29 +15,386 @@ input_data:
...
+      end_datetime: "2021-09-01T00:00:00"
 
   pv:
     pv_files_groups:
       - label: solar_sheffield_passiv
         pv_filename: gs://solar-pv-nowcasting-data/PV/Passive/ocf_formatted/v0/passiv.netcdf
         pv_metadata_filename: gs://solar-pv-nowcasting-data/PV/Passive/ocf_formatted/v0/system_metadata_OCF_ONLY.csv
+        # This is the list of pv_ml_ids to be sliced from the PV site level data
+    pv_ml_ids:
+      [
```
Comment (jacobbieker) [inline configs.example/datamodule/configuration/template_configuration.yaml:47]: I would suggest removing the MLIds from the example template, as it makes it a very long and large file.
```suggestion
      []
```

File: configs.example/datamodule/streamed_batches.yaml [hunk 4]
```diff
@@ -1,6 +1,8 @@
 _target_: pvnet.data.datamodule.DataModule
 # Path to the data configuration yaml file. You can find examples in the configuration subdirectory
 # in configs.example/datamodule/configuration
+# Use the full local path such as: /FULL/PATH/PVNet/configs/datamodule/configuration/gcp_configuration.yaml"
+num_workers: 20
```
Comment (jacobbieker) [inline configs.example/datamodule/streamed_batches.yaml:5]: pre-commit is failing on this right now, with duplicate keys, so might need to remove it
```suggestion
```

**Review by jacobbieker — CHANGES_REQUESTED** [review]

File: configs.example/datamodule/configuration/template_configuration.yaml (hunk 3 above)
Comment (peterdudfield) [reply configs.example/datamodule/configuration/template_configuration.yaml:47]: or leave in 10 examples

**Review by dfulu — APPROVED** [review]
Great work Zak!

File: requirements.txt [hunk 5]
```diff
@@ -23,3 +23,4 @@ hydra-core
 python-dotenv
 hydra-optuna-sweeper
 rich
+gcsfs
```
[the remark this answers is not in the document]
Comment (peterdudfield) [reply requirements.txt:26]: thanks

**Review by jacobbieker — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**